coordinator/history/fsstore.go

```diff
@@ -0,0 +1,50 @@
+// Copyright 2024 Edgeless Systems GmbH
+// SPDX-License-Identifier: AGPL-3.0-only
+
+package history
+
+import (
+	"bytes"
+	"fmt"
+	"path/filepath"
+	"sync"
+
+	"github.com/spf13/afero"
+)
+
+type fsStore struct {
+	fs  *afero.Afero
+	mux sync.RWMutex
+}
+
+func newPVStore(fs *afero.Afero) *fsStore {
+	return &fsStore{fs: fs}
+}
+
+func (s *fsStore) Get(key string) ([]byte, error) {
+	s.mux.RLock()
+	defer s.mux.RUnlock()
+	return s.fs.ReadFile(key)
+}
+
+func (s *fsStore) Set(key string, value []byte) error {
+	s.mux.Lock()
+	defer s.mux.Unlock()
+	if err := s.fs.MkdirAll(filepath.Base(key), 0o755); err != nil {
+		return fmt.Errorf("creating directory for %q: %w", key, err)
+	}
+	return s.fs.WriteFile(key, value, 0o644)
+}
+
+func (s *fsStore) CompareAndSwap(key string, oldVal, newVal []byte) error {
+	s.mux.Lock()
+	defer s.mux.Unlock()
+	current, err := s.Get(key)
```

Comment:
The `s.Get()` call also wants to acquire the same mutex. The easiest solution would be to inline the `s.Get()` and `s.Set()` call below (minus their mutex handling).